### PR TITLE
OCP27627

### PR DESCRIPTION
**Pull request by miyadav**
This commit implements - https://bugzilla.redhat.com/show_bug.cgi?id=1796599 test.
Polarion TEST case - OCP-27627
https://openshift-qe-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/Runner-v3/70903/console
https://openshift-qe-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/Runner-v3/70913/console

### CODE REVIEW

File: features/machine/machine.feature [hunk 1]
```diff
@@ -68,3 +68,31 @@ Feature: Machine features testing
       | https://machine-api-operator.openshift-machine-api.svc:8443/metrics          | # @case_id OCP-25652
       | https://cluster-autoscaler-operator.openshift-machine-api.svc:9192/metrics   | # @case_id OCP-26111
       | https://machine-approver.openshift-cluster-machine-approver.svc:9192/metrics | # @case_id OCP-26102
+
+
+
+  # @author miyadav@redhat.com
+  # @case_id OCP-27627
```
Comment (jhou1) [inline features/machine/machine.feature]: The two lines are not aligned with the following lines. Please use the same indention.

File: features/machine/machine.feature [hunk 2]
```diff
@@ -68,3 +68,31 @@ Feature: Machine features testing
...
+
+ @admin
+ Scenario: Verify all machine instance-state should be  consistent with their providerStats.instanceState
+   Given I have an IPI deployment
+   When I run the :get admin command with:
+      | resource  | machines                           |
+      | namespace | openshift-machine-api |
+      | o         | jsonpath='{range .items[*]}{.metadata.annotations}{.status.providerStatus.instanceState}{"/\\n"}{end}'|
```
Comment (jhou1) [inline features/machine/machine.feature]: The 3-line tables need to be aligned, i.e. the tailing `|` of each line should be aligned. There are some tools to auto-align such tables. Please also refer to https://github.com/openshift/verification-tests/blob/master/doc/vim.adoc

File: features/machine/machine.feature [hunk 3]
```diff
@@ -68,3 +68,31 @@ Feature: Machine features testing
...
       | https://cluster-autoscaler-operator.openshift-machine-api.svc:9192/metrics   | # @case_id OCP-26111
       | https://machine-approver.openshift-cluster-machine-approver.svc:9192/metrics | # @case_id OCP-26102
+
+
+
+  # @author miyadav@redhat.com
+  # @case_id OCP-27627
+
+ @admin
+ Scenario: Verify all machine instance-state should be  consistent with their providerStats.instanceState
```
Comment (jhou1) [inline features/machine/machine.feature]: There is a redundant space between "be  consistent"

File: features/machine/machine.feature [hunk 4]
```diff
@@ -68,3 +68,31 @@ Feature: Machine features testing
...
+
+   #Evaluating Result for each line
+   When I repeat the following steps for each :console_output in cb.console_output_array:
+
+   """
+   And evaluation of `cb.console_output_array.pop.split("]")[0]` is stored in the :provider_status clipboard
+   And evaluation of `cb.console_output_array.pop.split(":").pop.split("]")[1].split("/")[0]` is stored in the :instance_state clipboard
+
+   Then the expression should be true> cb.provider_status == cb.instance_state
+
```
Comment (jhou1) [inline features/machine/machine.feature]: Remove empty line

File: features/machine/machineProviderStatus_instanceState.feature [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+When I repeat the following steps for each :console_output in cb.console_output_array:
+
+   """
+
+   And evaluation of `cb.console_output.split(":")[1]` is stored in the :provider_status clipboard
+   And evaluation of `cb.console_output.split("]")[1]` is stored in the :instance_status clipboard
+
+   Then the expression should be true> cb.providerStatus == cb.instanceStatus
+
+   """
```
Comment (jhou1) [inline features/machine/machineProviderStatus_instanceState.feature]: Please keep these quoted lines same indent with `When I repeat` line

**Comment by jhou1** [comment]
@miyadav You have committed two files that cover a same test, can you please remove the file features/machine/machineProviderStatus_instanceState.feature and rebase your commit messages?

File: features/machine/machine.feature (hunk 2 above)
Comment (miyadav) [reply features/machine/machine.feature]: done

File: features/machine/machine.feature (hunk 3 above)
Comment (miyadav) [reply features/machine/machine.feature]: removed that

File: features/machine/machine.feature (hunk 4 above)
Comment (miyadav) [reply features/machine/machine.feature]: done

File: features/machine/machineProviderStatus_instanceState.feature (hunk 5 above)
Comment (miyadav) [reply features/machine/machineProviderStatus_instanceState.feature]: done

**Comment by akostadinov** [comment]
can be merged once @jhou1 concerns are adressed

File: features/machine/machine.feature [hunk 6]
```diff
@@ -68,3 +68,27 @@ Feature: Machine features testing
...
+   And evaluation of `@result[:response].split(":")` is stored in the :console_output_array clipboard
+
+   #Evaluating Result for each line
+   When I repeat the following steps for each :console_output in cb.console_output_array:
+   """
+   And evaluation of `cb.console_output_array.pop.split("]")[0]` is stored in the :provider_status clipboard
+   And evaluation of `cb.console_output_array.pop.split(":").pop.split("]")[1].split("/")[0]` is stored in the :instance_state clipboard
+   Then the expression should be true> cb.provider_status == cb.instance_state
+   """
+  Then the step should succeed
```
Comment (jhou1) [inline features/machine/machine.feature]: This line needs to be aligned to the `When` clause

**Comment by jhou1** [comment]
Now almost done, can you please squash the commit messages into one?

File: features/machine/machine.feature (hunk 6 above)
Comment (miyadav) [reply features/machine/machine.feature]: done

**Comment by miyadav** [comment]
Due to commits messages which were not clear and might not be good for master closing this pull request and will open a new one

**Pull request closed without merging**